mbuild/formats/lammpsdata.py

```diff
@@ -524,3 +488,121 @@ def write_lammpsdata(structure, filename, atom_style='full',
+                                                     round(dihedral.type.scee,1),
+                                                     round(dihedral.type.scnb,1),
+                                                     dihedral.atom1.type, dihedral.atom2.type,
+                                                     dihedral.atom3.type, dihedral.atom4.type
+                                                     ) for dihedral in structure.rb_torsions])))
+        unique_dihedral_types = OrderedDict([(y,x+1) for x,y in unique_dihedral_types.items()])
+        dihedral_types = [unique_dihedral_types[(round(dihedral.type.c0*lj_unit,3),
+                                                 round(dihedral.type.c1*lj_unit,3),
+                                                 round(dihedral.type.c2*lj_unit,3),
+                                                 round(dihedral.type.c3*lj_unit,3),
+                                                 round(dihedral.type.c4*lj_unit,3),
+                                                 round(dihedral.type.c5*lj_unit,3),
+                                                 round(dihedral.type.scee,1),
+                                                 round(dihedral.type.scnb,1),
+                                                 dihedral.atom1.type, dihedral.atom2.type,
+                                                 dihedral.atom3.type, dihedral.atom4.type
+                                                 )] for dihedral in structure.rb_torsions]
+        
+    elif use_dihedrals:
+        charmm_dihedrals = []
+        structure.join_dihedrals()
+        for dihedral in structure.dihedrals:
+            if not dihedral.improper:
+                weight = 1 / len(dihedral.type)
+                for dih_type in dihedral.type:
+                    charmm_dihedrals.append((round(dih_type.phi_k,3),
+                                             int(round(dih_type.per,0)),
+                                             int(round(dih_type.phase,0)),
+                                             round(weight, 4),
+                                             round(dih_type.scee,1),
+                                             round(dih_type.scnb,1),
+                                             dihedral.atom1.type, dihedral.atom2.type,
+                                             dihedral.atom3.type, dihedral.atom4.type))
+
+        unique_dihedral_types = dict(enumerate(set(charmm_dihedrals)))
+        unique_dihedral_types = OrderedDict([(y,x+1) for x,y in unique_dihedral_types.items()])
+        dihedral_types = [unique_dihedral_types[dihedral_info] for dihedral_info in charmm_dihedrals]
+
+    return dihedral_types, unique_dihedral_types
+
+def _get_impropers(structure, sigma_conversion_factor, epsilon_conversion_factor):
+    lj_unit = sigma_conversion_factor / epsilon_conversion_factor
+    unique_improper_types = dict(enumerate(set([(round(improper.type.psi_k*lj_unit,3),
```

Comment:
For [harmonic impropers](https://lammps.sandia.gov/doc/improper_harmonic.html), looks like `psi_k` is in units of energy/radians, so we don't need the sigma conversion factor